File: scripts/debug/auth/test_session_loading.py
#!/usr/bin/env python3
"""Test session loading after OAuth callback"""
from playwright.sync_api import sync_playwright
import json

print("🧪 Testing session loading with Playwright...")

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    context = browser.new_context()
    page = context.new_page()

    # Capture console logs
    console_logs = []
    def handle_console(msg):
        console_logs.append(f"[{msg.type}] {msg.text}")
    page.on("console", handle_console)

    # Capture network requests and responses
    requests = []
    def handle_request(request):
        requests.append({
            'method': request.method,
            'url': request.url,
            'headers': dict(request.headers)
        })
    page.on("request", handle_request)

    responses = []
    def handle_response(response):
        if response.status >= 400:
            responses.append({
                'url': response.url,
                'status': response.status,
                'statusText': response.status_text
            })
    page.on("response", handle_response)

    # Navigate to the page after OAuth callback
    print("\n📍 Loading http://localhost:3000/?auth=ok")
    page.goto('http://localhost:3000/?auth=ok')

    # Wait for network idle
    print("⏳ Waiting for networkidle...")
    page.wait_for_load_state('networkidle', timeout=10000)

    # Take screenshot
    print("📸 Taking screenshot...")
    page.screenshot(path='/tmp/session-loading.png', full_page=True)

    # Wait a bit more for async operations
    page.wait_for_timeout(2000)

    # Check if session loaded
    print("\n🔍 Checking page content...")
    content = page.content()
    if "Lade Session" in content:
        print("❌ Session still loading (stuck on 'Lade Session ...')")
    else:
        print("✅ Session text changed (may have loaded)")

    # Print console logs
    print(f"\n📝 Console logs ({len(console_logs)} entries):")
    for log in console_logs:
        print(f"  {log}")

    # Check for /auth/me request
    print(f"\n🌐 Network requests ({len(requests)} total):")
    auth_me_requests = [r for r in requests if '/auth/me' in r['url']]
    if auth_me_requests:
        print(f"  ✅ Found {len(auth_me_requests)} request(s) to /auth/me")
        for req in auth_me_requests:
            print(f"    Method: {req['method']}")
            print(f"    Cookie: {req['headers'].get('cookie', '(none)')}")
    else:
        print("  ❌ No request to /auth/me found")

    browser.close()
    print("\n✅ Test complete. Screenshot: /tmp/session-loading.png")

File: scripts/debug/auth/test_session_loading.ts
import { chromium, type ConsoleMessage, type Request, type Response } from 'playwright';

interface CapturedRequest {
  method: string;
  url: string;
  headers: Record<string, string>;
}

interface CapturedResponse {
  url: string;
  status: number;
  statusText: string;
}

const toConsoleLine = (message: ConsoleMessage): string => `[${message.type()}] ${message.text()}`;

const toCapturedRequest = (request: Request): CapturedRequest => ({
  method: request.method(),
  url: request.url(),
  headers: request.headers(),
});

const toCapturedResponse = (response: Response): CapturedResponse => ({
  url: response.url(),
  status: response.status(),
  statusText: response.statusText(),
});

const run = async (): Promise<void> => {
  console.log('Testing session loading with Playwright...');

  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();

  const consoleLogs: string[] = [];
  const requests: CapturedRequest[] = [];
  const responses: CapturedResponse[] = [];

  page.on('console', (message) => {
    consoleLogs.push(toConsoleLine(message));
  });

  page.on('request', (request) => {
    requests.push(toCapturedRequest(request));
  });

  page.on('response', (response) => {
    if (response.status() >= 400) {
      responses.push(toCapturedResponse(response));
    }
  });

  console.log('\nLoading http://localhost:3000/?auth=ok');
  await page.goto('http://localhost:3000/?auth=ok');

  console.log('Waiting for networkidle...');
  await page.waitForLoadState('networkidle', { timeout: 10_000 });

  console.log('Taking screenshot...');
  await page.screenshot({ path: '/tmp/session-loading.png', fullPage: true });
  await page.waitForTimeout(2000);

  console.log('\nChecking page content...');
  const content = await page.content();
  if (content.includes('Lade Session')) {
    console.log("Session still loading (stuck on 'Lade Session ...')");
  } else {
    console.log('Session text changed (may have loaded)');
  }

  console.log(`\nConsole logs (${consoleLogs.length} entries):`);
  for (const log of consoleLogs) {
    console.log(`  ${log}`);
  }

  console.log(`\nNetwork requests (${requests.length} total):`);
  const authRequests = requests.filter((request) => request.url.includes('/auth/me'));
  if (authRequests.length > 0) {
    console.log(`  Found ${authRequests.length} request(s) to /auth/me`);
    for (const request of authRequests) {
      console.log(`    Method: ${request.method}`);
      console.log(`    Cookie: ${request.headers.cookie ?? '(none)'}`);
    }
  } else {
    console.log('  No request to /auth/me found');
  }

  if (responses.length > 0) {
    console.log(`\nFailing responses (${responses.length}):`);
    for (const response of responses) {
      console.log(`  ${response.status} ${response.statusText} ${response.url}`);
    }
  }

  await browser.close();
  console.log('\nTest complete. Screenshot: /tmp/session-loading.png');
};

void run();
